src/test/About/About.test.tsx

```diff
@@ -35,29 +35,49 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
  * SOFTWARE.
  */
-import React from 'react';
-import { App } from '@app/index';
-import { mount, shallow } from 'enzyme';
-import { Button } from '@patternfly/react-core';
+import * as React from 'react';
+import renderer from 'react-test-renderer'
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import { CRYOSTAT_TRADEMARK } from '@app/About/AboutDescription';
+import { About } from '@app/About/About';
 
-describe('App tests', () => {
-  test('should render default App component', () => {
-    const view = shallow(<App />);
-    expect(view).toMatchSnapshot();
-  });
+jest.mock('@app/BreadcrumbPage/BreadcrumbPage', () => {
+  return {
+    BreadcrumbPage: jest.fn((props) => {
+      return <div>
+                {props.pageTitle}
+                {props.children}
+             </div>
+    })
+  };
+});
 
-  it('should render a nav-toggle button', () => {
-    const wrapper = mount(<App />);
-    const button = wrapper.find(Button);
-    expect(button.exists()).toBe(true);
+jest.mock('@app/About/AboutDescription', () => {
+  return {
+    ...jest.requireActual('@app/About/AboutDescription'),
+    AboutDescription: jest.fn(() => {
+      return <div>
+                AboutDescription
+             </div>
+    })
+  };
+});
+
+describe('<About />', () => {
+  it('renders correctly', () => {
+    const tree = renderer.create(<About />);
+    expect(tree.toJSON()).toMatchSnapshot();
   });
 
-  it('should hide the sidebar when clicking the nav-toggle button', () => {
-    const wrapper = mount(<App />);
-    const button = wrapper.find('#nav-toggle').hostNodes();
-    expect(wrapper.find('#page-sidebar').hasClass('pf-m-expanded')).toBeTruthy();
-    button.simulate('click');
-    expect(wrapper.find('#page-sidebar').hasClass('pf-m-collapsed')).toBeTruthy();
-    expect(wrapper.find('#page-sidebar').hasClass('pf-m-expanded')).toBeFalsy();
+  it('contains the correct information', () => {
```

Comment:
I was wondering how you decided what kind of information should be asserted here. How did you know which DOM elements/props to test here that aren't already covered in the snapshot test?

Reply:
After the `render` call in a specific unit test, insert the line `screen.debug()` and re-run the test suite. What this does is output the render result (i.e. the HTML DOM representation) to the command line so you'll be able to see why my queries are structured the way they are.

Reply:
For the information itself, I tested to see whether the props that are passed into the child components inside the `About` component show up as intended. For example, I test for the `CRYOSTAT_TRADEMARK` which is passed into the body of the `CardFooter` component (line 55 in `About.tsx`). However, there is some guess-and-check required to figure out how this information is represented in the actual DOM.

Reply:
Oh I think I get it now - the snapshot tests check that the generated HTML looks the same as before, while these unit tests check that the HTML rendered in the DOM actually have the expected behaviour